File: Java/Nonmobile/Abstract_Classes/Prostokat.java
package com.example.javalib;

public class Prostokat extends Figura{

    private double b;

    public Prostokat(double a, double b)
    {
        super(a);
        this.b = b;
    }
    public double Pole()
    {
        return a * b;
    }

    public double Obwod()
    {
        return a * 2 + b * 2;
    }

    public void oFigurze()
    {
        System.out.println("Jestem prostokątem");
        super.oFigurze();
    }
    
}

File: C#/Abstract_Classes/KlasyAbstrakcyjne/Prostokat.cs
namespace KlasyAbstrakcyjne;

public class Prostokat : Figura
{
    
    // BEZ DZIEDZICZENIA KONSTRUKTORA
    // private double _bok1;
    // private double _bok2;
    
    // public Prostokat(double bok1, double bok2)
    // {
    //     _bok1 = bok1;
    //     _bok2 = bok2;
    // }
    // public override double Pole()
    // {
    //     return _bok1 * _bok2;
    // }
    //
    // public override double Obwod()
    // {
    //     return _bok1 * 2 + _bok2 * 2;
    // }

    // Z DZIEDZICZENIEM KONSTRUKTORA
    private double _b;
    public Prostokat(double a, double b) : base(a)
    {
        _b = b;
    }
    public override double Pole()
    {
        return a * _b;
    }
    
    public override double Obwod()
    {
        return a * 2 + _b * 2;
    }
    
    public override void oFigurze()
    {
        Console.WriteLine("Jestem prostokątem");
        base.oFigurze();
    }
}